stock_request/models/stock_request_abstract.py

```diff
@@ -107,20 +107,26 @@ def _compute_product_qty(self):
     @api.depends("product_id", "warehouse_id", "location_id")
     def _compute_route_ids(self):
         route_obj = self.env["stock.location.route"]
-        for wh in self.mapped("warehouse_id"):
-            wh_routes = route_obj.search([("warehouse_ids", "=", wh.id)])
-            for record in self.filtered(lambda r: r.warehouse_id == wh):
-                routes = route_obj
-                if record.product_id:
-                    routes += record.product_id.mapped(
-                        "route_ids"
-                    ) | record.product_id.mapped("categ_id").mapped("total_route_ids")
-                if record.warehouse_id:
-                    routes |= wh_routes
-                parents = record.get_parents().ids
-                record.route_ids = routes.filtered(
-                    lambda r: any(p.location_id.id in parents for p in r.rule_ids)
+        routes = route_obj.search([("warehouse_ids", "!=", False)])
```

Comment:
Maybe this will get too much records. If I understand well, only interested routes should be self.mapped("warehouse_id").
So, I would have done :

```suggestion
        routes = route_obj.search([("warehouse_ids", "in", self.mapped("warehouse_id"))])
```

Reply:
it makes sense

Reply:
done, change by `self.mapped("warehouse_id").ids`